RecoLocalCalo/EcalRecProducers/python/ecalUncalibRecHitPhase2_cff.py

```diff
@@ -1,4 +1,55 @@
 import FWCore.ParameterSet.Config as cms
+from HeterogeneousCore.CUDACore.SwitchProducerCUDA import SwitchProducerCUDA
+from Configuration.ProcessModifiers.gpu_cff import gpu
 
-from RecoLocalCalo.EcalRecProducers.ecalUncalibRecHitPhase2_cfi import *
+
+from RecoLocalCalo.EcalRecProducers.ecalUncalibRecHitPhase2_cfi import ecalUncalibRecHitPhase2 as _ecalUncalibRecHitPhase2
+ecalUncalibRecHitPhase2 = SwitchProducerCUDA(
+  cpu = _ecalUncalibRecHitPhase2.clone()
+)
+
+# cpu weights
 ecalUncalibRecHitPhase2Task = cms.Task(ecalUncalibRecHitPhase2)
+
+# conditions used on gpu
+
+
+from RecoLocalCalo.EcalRecProducers.ecalPhase2DigiToGPUProducer_cfi import ecalPhase2DigiToGPUProducer as _ecalPhase2DigiToGPUProducer
+ecalPhase2DigiToGPUProducer = _ecalPhase2DigiToGPUProducer.clone()
+
+# gpu weights
+from RecoLocalCalo.EcalRecProducers.ecalUncalibRecHitPhase2GPU_cfi import ecalUncalibRecHitPhase2GPU as _ecalUncalibRecHitPhase2GPU
+ecalUncalibRecHitPhase2GPU = _ecalUncalibRecHitPhase2GPU.clone(
+  digisLabelEB = ('ecalPhase2DigiToGPUProducer', 'ebDigis')
+)
+
+# copy the uncalibrated rechits from GPU to CPU
+from RecoLocalCalo.EcalRecProducers.ecalCPUUncalibRecHitProducer_cfi import ecalCPUUncalibRecHitProducer as _ecalCPUUncalibRecHitProducer
+ecalUncalibRecHitSoA = _ecalCPUUncalibRecHitProducer.clone(
+  recHitsInLabelEB = ('ecalUncalibRecHitPhase2GPU', 'EcalUncalibRecHitsEB'),
+  isPhase2 = True,
+  recHitsInLabelEE = None,  # remove unneeded Phase1 parameters
+  recHitsOutLabelEE = None
+)
+
+
+from RecoLocalCalo.EcalRecProducers.ecalUncalibRecHitConvertGPU2CPUFormat_cfi import ecalUncalibRecHitConvertGPU2CPUFormat as _ecalUncalibRecHitConvertGPU2CPUFormat
+gpu.toModify(ecalUncalibRecHitPhase2,
+    cuda = _ecalUncalibRecHitConvertGPU2CPUFormat.clone(
+        isPhase2 = cms.bool(True),
+        recHitsLabelGPUEB = cms.InputTag('ecalUncalibRecHitSoA', 'EcalUncalibRecHitsEB'),
```

Comment:
I'll just note here that 
```suggestion
        isPhase2 = True,
        recHitsLabelGPUEB = ('ecalUncalibRecHitSoA', 'EcalUncalibRecHitsEB'),
```
would be preferred, but this can be done as a quick follow-up too.